Gems/PhysX/Code/Tests/EditorColliderTests.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzToolsFramework/ToolsComponents/EditorNonUniformScaleComponent.h>
+#include <AzToolsFramework/UnitTest/AzToolsFrameworkTestHelpers.h>
+#include <EditorColliderComponent.h>
+#include <EditorForceRegionComponent.h>
+#include <EditorRigidBodyComponent.h>
+#include <EditorShapeColliderComponent.h>
+#include <LmbrCentral/Shape/BoxShapeComponentBus.h>
+#include <LmbrCentral/Shape/CompoundShapeComponentBus.h>
+#include <LmbrCentral/Shape/CylinderShapeComponentBus.h>
+#include <LmbrCentral/Shape/PolygonPrismShapeComponentBus.h>
+#include <LmbrCentral/Shape/QuadShapeComponentBus.h>
+#include <LmbrCentral/Shape/SphereShapeComponentBus.h>
+#include <PhysX/ForceRegionComponentBus.h>
+#include <PhysX/MathConversion.h>
+#include <PhysX/PhysXLocks.h>
+#include <PhysX/SystemComponentBus.h>
+#include <RigidBodyComponent.h>
+#include <RigidBodyStatic.h>
+#include <ShapeColliderComponent.h>
+#include <StaticRigidBodyComponent.h>
+#include <Tests/EditorTestUtilities.h>
+#include <Tests/PhysXTestCommon.h>
+
+namespace PhysXEditorTests
+{
+    namespace
+    {
+        struct TestData
+        {
+            const AZStd::vector<AZ::Vector2> polygonHShape = { AZ::Vector2(0.0f, 0.0f), AZ::Vector2(0.0f, 3.0f), AZ::Vector2(1.0f, 3.0f),
+                                                               AZ::Vector2(1.0f, 2.0f), AZ::Vector2(2.0f, 2.0f), AZ::Vector2(2.0f, 3.0f),
+                                                               AZ::Vector2(3.0f, 3.0f), AZ::Vector2(3.0f, 0.0f), AZ::Vector2(2.0f, 0.0f),
+                                                               AZ::Vector2(2.0f, 1.0f), AZ::Vector2(1.0f, 1.0f), AZ::Vector2(1.0f, 0.0f) };
+        };
+    } // namespace
+
+    TEST_F(PhysXEditorFixture, EditorShapeColliderComponent_ShapeDependencySatisfied_EntityIsValid)
```

Comment:
update: Looks like the test name needs to be updated 😉 Looks like a great start though! Give me or @greerdv/@jonawals a ping when this is ready for review 👍